src/accelerate/state.py

```diff
@@ -54,6 +55,21 @@ def do_nothing(*args, **kwargs):
     return None
 
 
+class GlobalSharedDict:
+  def __init__(self, thread_local: bool = False):
+    self._storage = {}
+
+  def __get__(self, obj, objtype=None):
+    return self._storage
+
+  def __set__(self, obj, value):
+    self._storage = value
+
+class ThreadLocalSharedDict(GlobalSharedDict, threading.local):
+  pass
+
+SharedDict = GlobalSharedDict if not is_tpu_available(check_device=False) else ThreadLocalSharedDict
```

Comment:
Let's not break the existing ang keep using a regular dict when not on TPU.

Reply:
Leaving a dict and calling clear seems easier to me, as long as it works. We don't have to worry about the state being multithreaded usually (as it's just launched on different processes) so I believe this is the first time it's coming up.

Reply:
Done.